qulice-checkstyle/src/main/java/com/qulice/checkstyle/JavadocParameterOrderCheck.java

```diff
@@ -30,36 +30,216 @@
 package com.qulice.checkstyle;
 
 import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.TextBlock;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTag;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * Checks method parameters order to comply with what is defined in method
  * javadoc.
  *
  * @since 0.18.10
- * @todo #847:30min Enforce right parameter order in javadoc. Checkstyle must
- *  check if parameter order in javadoc is the same as is in the method
- *  signature. Implement this check and add its tests to checks.xml under
- *  custom checks.
  */
+@SuppressWarnings({"PMD.AvoidInstantiatingObjectsInLoops", "PMD.LongVariable"})
 public final class JavadocParameterOrderCheck extends AbstractCheck {
+
+    /**
+     * Compiled regexp to match Javadoc tags that take an argument.
+     */
+    private static final Pattern MATCH_JAVADOC_ARG = Pattern.compile(
+        "^\\s*(?>\\*|\\/\\*\\*)?\\s*@(param)\\s+(\\S+)\\s+\\S*", 0
```

Comment:
@paulodamaso the `flags = 0` is default for single argument Pattern.compile

Reply:
@krzyk Commited using no parameters, let's see what happens